### PR TITLE
[proof] optimize proof always works on incomplete proofs

### LINKED ISSUE #9451: Anomaly "grounding a non evar-free term" during Optimize Proof
#### Version

bad 4b006833d6f866a33024e674d300f74592d24622, good 8.9.0

#### Operating system

Arch Linux, up to date as of yesterday.

#### Description of the problem

```coq
Goal False. evar (x : True). Optimize Proof. Abort. 
(* Error: Anomaly "grounding a non evar-free term" *)
```

```coq
Goal False.
cut True.
assert False.
evar (x : True).
let v := open_constr:(_) in idtac. all: exfalso; clear.
Optimize Proof.
(* Error: Anomaly "grounding a non evar-free term" *)
Abort All.
```

Original finding: adding `Optimize Proof` on this bedrock2 line https://github.com/mit-plv/bedrock2/commit/48e2314623e8695df7f0bf2e390ce224baf7a161#diff-5c7e120c9281033af09d97fd489aa26aR74, where the line after which Optimize Proof starts anomalying is a call to [seprewrite](https://github.com/mit-plv/bedrock2/blob/master/bedrock2/src/Map/SeparationLogic.v#L228).

### PR DESCRIPTION
**Pull request by gares**
Fix #9451

### CODE REVIEW

File: engine/proofview.ml [hunk 1]
```diff
@@ -39,7 +39,7 @@ let proofview p =
 
 let compact el ({ solution } as pv) =
   let nf c = Evarutil.nf_evar solution c in
-  let nf0 c = EConstr.(to_constr solution (of_constr c)) in
```
Comment (ejgallego) [inline engine/proofview.ml]: Why is this not using the standard `nf` functions?

**Review by JasonGross — DISMISSED** [review]
Add a file to the test suite to ensure this doesn't happen again?

File: engine/proofview.ml (hunk 1 above)
Comment (gares) [reply engine/proofview.ml]: I've no idea, when I wrote this code EConstr did not even exist, not to talk about the anomaly about ground terms...

Comment (gares) [reply engine/proofview.ml]: It comes from c8c8ccdaaffefdbd3d78c844552a08bcb7b4f915

**Review by ppedrot — APPROVED** [review]
LGTM. This is only side-stepping the problem though.

File: engine/proofview.ml (hunk 1 above)
Comment (SkySkimmer) [reply engine/proofview.ml]: nf_evars on constr is the awkwardly-named nf_evars_universes

**Pull request merged**